lib/AST/ASTImporter.cpp

```diff
@@ -3291,9 +3293,11 @@ Decl *ASTNodeImporter::VisitVarDecl(VarDecl *D) {
   ToVar->setAccess(D->getAccess());
   ToVar->setLexicalDeclContext(LexicalDC);
 
-  // Templated declarations should never appear in the enclosing DeclContext.
-  if (!D->getDescribedVarTemplate())
-    LexicalDC->addDeclInternal(ToVar);
+  if (MergeWithVar) {
+    auto *Recent = const_cast<VarDecl *>(MergeWithVar->getMostRecentDecl());
+    updateAttrAndFlags(D, MergeWithVar);
```

Comment:
The `updateAttrAndFlags` is used to update data (from `D`) in the found Decl. I think this should be called at the `MapImported` cases (line 3223 and 3229).

Reply:
Ok, changed that.

Reply:
This will not work that way. Just tried and moved `updateAttrAndFlags` before the MapImported calls, and the unit test which test the used flag fails.
If we think about it this is logical, since we should update the used flag when we do create a new VarDecl, and that is happening once we reach to this line. 
In those lines you mentioned (line 3223 and 3229) we do not create any new declaration, thus there is nothing to update.

Reply:
Okay, makes sense, I am going to check that and also the test.

Reply:
So, I've written two new test cases which exercises those cases. And indeed it is needed to add `updateAttrAndFlags` there.
Then I realized that we have to do this always after we found an existing decl or even if we created a new one.
This seems like a general rule, so I added the call of `updateAttrAndFlags` into `Import::(Decl*)`.